src/common/wal/src/config/kafka/datanode.rs

```diff
@@ -39,6 +45,105 @@ pub struct DatanodeKafkaConfig {
     /// The kafka topic config.
     #[serde(flatten)]
     pub kafka_topic: KafkaTopicConfig,
+    /// Client SASL.
+    pub sasl: Option<KafkaClientSasl>,
+    /// Client TLS config
+    pub tls: Option<KafkaClientTls>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct KafkaClientSasl {
+    #[serde(flatten)]
+    pub config: KafkaClientSaslConfig,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+#[serde(tag = "type", rename_all = "SCREAMING-KEBAB-CASE")]
+pub enum KafkaClientSaslConfig {
+    Plain {
+        username: String,
+        password: String,
+    },
+    #[serde(rename = "SCRAM-SHA-256")]
+    ScramSha256 {
+        username: String,
+        password: String,
+    },
+    #[serde(rename = "SCRAM-SHA-512")]
+    ScramSha512 {
+        username: String,
+        password: String,
+    },
+}
+
+impl KafkaClientSaslConfig {
+    /// Converts to [`SaslConfig`].
+    pub fn into_sasl_config(self) -> SaslConfig {
+        match self {
+            KafkaClientSaslConfig::Plain { username, password } => {
+                SaslConfig::Plain(Credentials::new(username, password))
+            }
+            KafkaClientSaslConfig::ScramSha256 { username, password } => {
+                SaslConfig::ScramSha256(Credentials::new(username, password))
+            }
+            KafkaClientSaslConfig::ScramSha512 { username, password } => {
+                SaslConfig::ScramSha512(Credentials::new(username, password))
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq)]
+pub struct KafkaClientTls {
+    pub server_ca_cert_path: String,
+    pub client_cert_path: Option<String>,
+    pub client_key_path: Option<String>,
+}
+
+impl KafkaClientTls {
+    /// Builds the [`ClientConfig`].
+    pub fn to_tsl_config(&self) -> Result<Arc<ClientConfig>> {
+        let builder = ClientConfig::builder();
+        let mut roots = RootCertStore::empty();
+
+        let root_cert_bytes =
+            std::fs::read(&self.server_ca_cert_path).context(error::ReadFileSnafu {
+                path: &self.server_ca_cert_path,
+            })?;
```

Comment:
Avoid synchronous IO in async context. Looks like `KafkaClientTls::to_tsl_config` is invoked from `ClientManager::try_new`.